### PR TITLE
Return null from FakeDynamo on upsert

### PR DESCRIPTION
**Pull request by chaosgame**
Hello @xiao, @sboora, @kylehg, 

Please review the following commits I made in branch 'nathan-perf'.

0278ef9d85273dc3c0ffad64f2d96bc1311b6193 (2015-09-08 12:00:51 -0700)
Return null from FakeDynamo on upsert

R=@xiao
R=@sboora
R=@kylehg

Test plan: 'Added checks to unit test'

### CODE REVIEW

File: lib/FakeDynamo.js [hunk 1]
```diff
@@ -360,7 +362,7 @@ FakeTable.prototype.updateItem = function(data) {
 
   // done (ALL_OLD only returns ConsumedCapacity)
   return Q.resolve({
-    Attributes: oldItem,
+    Attributes: itemExists ? oldItem : null,
```
Comment (sboora) [inline lib/FakeDynamo.js]: this might make it inconsistent with the actual API, does it?

Comment (chaosgame) [inline lib/FakeDynamo.js]: This actually fixes the consistency issue.  Previously we were returning `oldItem` which had been updated with values that shouldn't exist on lines 349-352.  The additional check in the unit test is being run against `FakeDynamo` to double check that this behavior is correct.

Comment (sboora) [inline lib/FakeDynamo.js]: :+1:

**Comment by sboora** [comment]
LG!

**Pull request merged**